.github/workflows/run-tests.yml

```
@@ -41,8 +48,9 @@ jobs:
             ~/.local/share/pypoetry
           key: ${{ runner.os }}-python-${{ matrix.version }}-poetry-${{ hashFiles('pyproject.toml', 'tox.ini') }}
       - name: Install Poetry
-        if: steps.cache-deps.outputs.cache-hit != 'true'
-        run: curl -sSL https://install.python-poetry.org | python3 -
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: 1.5.1
```

```suggestion
        if: steps.cache-deps.outputs.cache-hit != 'true' && ! startsWith (matrix.os, 'windows')
        run: curl -sSL https://install.python-poetry.org | python3 -
      - name: Install Poetry (Windows)
        if: steps.cache-deps.outputs.cache-hit != 'true' && startsWith (matrix.os, 'windows')
        shell: pwsh
        run: |
          (Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | py -
```

I'm personally not a huge fan of 3rd party Actions since there's no real auditing or verification. Also I note for the selected action, they are installing Poetry with `pip` which is not recommended.

According to GitHub Actions own documentation, we can just automatically [use PowerShell in an Action](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-powershell), so I'm wondering if we could just conditionally use PowerShell to install Poetry for the Windows runners?